apps/src/templates/VersionHistory.jsx

```diff
@@ -0,0 +1,92 @@
+var React = require('react');
+var VersionRow = require('./VersionRow.jsx');
+var sourcesApi = require('../clientApi')('sources');
+
+/**
+ * A component for viewing project version history.
+ */
+module.exports = React.createClass({
+
+  getInitialState: function () {
+    return {
+      versions: null,
+      statusMessage: ''
+    };
+  },
+
+  componentWillMount: function () {
+    // TODO: Use Dave's client api when it's finished.
+    sourcesApi.ajax('GET', 'main.json/versions', this.onVersionListReceived, this.onAjaxFailure);
```

Comment:
It appears from searching around a bit that it is RESTful (1) to have collections within resources (this was surprising to me) and (2) for resources to have '.' in their names. So based on that I'm having a hard time arguing against main.json/foo, even though it does seem weird.